app/src/main/java/net/squanchy/notification/NotificationsIntentService.java

```diff
@@ -45,23 +49,33 @@ public void onCreate() {
     }
 
     @Override
+    public void onStart(@Nullable Intent intent, int startId) {
+        super.onStart(intent, startId);
+        subscriptions = new CompositeDisposable();
+    }
+
+    @Override
     protected void onHandleIntent(Intent intent) {
         LocalDateTime notificationIntervalEnd = new LocalDateTime().plusMinutes(NOTIFICATION_INTERVAL_MINUTES);
 
         Observable<List<Event>> sortedFavourites = service.sortedFavourites();
 
         LocalDateTime now = new LocalDateTime();
         if (shouldShowNotifications()) {
-            sortedFavourites
-                    .map(events -> filter(events, event -> event.getStartTime().isAfter(now)))
-                    .map(events -> filter(events, event -> isBeforeOrEqualTo(event.getStartTime(), notificationIntervalEnd)))
-                    .map(notificationCreator::createFrom)
-                    .subscribe(notifier::showNotifications);
+            subscriptions.add(
```

Comment:
Ok it's as I wanted it to be then, I don't think we should clear before the if. The `IntentService` will process intents sequentially, and we're not specifying an `observeOn()`, so it wouldn't make much of a difference. Still, I have put it in `onDestroy` which seems the best place for that (note that `IntentService` does not have `onStop`). See 8e172f5cf921b19408d3a311834bd02a9a677831